kuiper-tests/kuiper-tests.csproj

```diff
@@ -9,6 +9,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.4" />
+    <PackageReference Include="moq" Version="4.16.1" />
```

Comment:
I suppose the first one to make main gets to choose mocking framework :D I have set mine up to NSub but i am not religous.

Reply:
Neither am I, it was the first thing I stumbled upon when looking for mock frameworks for xunit. 🤷

Reply:
I am good with both Moq and NSubstitute, they're very similar in usage.

I think the only antsy gripe I have with Moq is the use of lambda for setup and verification/assertion, whereas NSub is more fluent and readable with a simpler .X syntax. However, if we can see past that, Moq has superior throws assertion and handling of matching input on mocks. 

I cast my vote for Moq.

Reply:
Moq it is then! A good opportunity to try that out

Reply:
Moq is superior..